cmake/Corrosion.cmake

```
@@ -703,6 +703,11 @@ function(_add_cargo_build out_cargo_build_out_dir)
         list(APPEND corrosion_cc_rs_flags "SDKROOT=${CMAKE_OSX_SYSROOT}")
     endif()
 
+    # Ensure that cc-rs targets same Apple platform version as the CMake build
+    if(CMAKE_HOST_SYSTEM_NAME STREQUAL "Darwin" AND CMAKE_OSX_DEPLOYMENT_TARGET)
```

```suggestion
    if(CMAKE_SYSTEM_NAME STREQUAL "Darwin" AND CMAKE_OSX_DEPLOYMENT_TARGET)
```

It should only matter that CMake is targetting a MacOS system, and not if the host system is macos, right?

---

I think you are right. Despite 'OSX' in the name CMake uses the `CMAKE_OSX_DEPLOYMENT_TARGET` variable for all Apple targets (also iOS, etc), but `cc-rs` is [sensitive](https://github.com/rust-lang/cc-rs/blob/218365e5837815124b4b30c0ebb8149e1cec092a/src/lib.rs#L3996) to the `IPHONEOS_DEPLOYMENT_TARGET` only if the target is actual macOS.